src/common/transformations/tests/op_conversions/convert_gather_to_compressed_test.cpp

```diff
@@ -147,8 +141,41 @@ TEST_F(TransformationTestsF, ConvertGatherToCompressed4) {
                                                                                       axis_const,
                                                                                       0,
                                                                                       scale_const,
-                                                                                      zp_const,
-                                                                                      ov::element::f32);
+                                                                                      zp_const);
+
+        model_ref = std::make_shared<ov::Model>(ov::NodeVector{gather_compressed}, ov::ParameterVector{input1});
+    }
+}
+
+TEST_F(TransformationTestsF, ConvertGatherToCompressed5) {
+    {
+        auto input1 = std::make_shared<ov::op::v0::Parameter>(ov::element::i32, ov::PartialShape{-1, 16});
+        auto axis_const = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{1}, {1});
+        auto weights_const = ov::op::v0::Constant::create(ov::element::u8, ov::Shape{32, 16}, {1});
+        auto convert = std::make_shared<ov::op::v0::Convert>(weights_const, ov::element::f32);
+        auto zp_const = ov::op::v0::Constant::create(ov::element::f32, ov::Shape{32, 1}, {1});
+        auto sub = std::make_shared<ov::op::v1::Subtract>(convert, zp_const);
+        auto scale_const = ov::op::v0::Constant::create(ov::element::f32, ov::Shape{32, 1}, {1});
+        auto scale = std::make_shared<ov::op::v1::Multiply>(sub, scale_const);
+        auto scale_convert = std::make_shared<ov::op::v0::Convert>(scale, ov::element::f32);
+        auto gather = std::make_shared<ov::op::v8::Gather>(scale_convert, input1, axis_const);
```

Comment:
The test case seems a bit confusing because `scale_convert` does actually nothing -- it has f32 precision on both input and output. I'd suggest creating decompression subgraph in f16 precision, and convert the subgraph to f32 by `scale_convert`. What do you think?

Reply:
For only testing transformation, current test could cover our new added code. For the real usage case, you are right, we need a real decomposing graph. Let me add it.